render-ws/src/main/java/org/janelia/render/service/CombinedRenderAndMatchService.java

```diff
@@ -0,0 +1,256 @@
+    public CombinedRenderAndMatchService()
+            throws UnknownHostException {
+        this(RenderDao.build(), MatchDao.build());
+    }
+
+    private CombinedRenderAndMatchService(final RenderDao renderDao,
+                                          final MatchDao matchDao) {
+        this.renderDao = renderDao;
+        this.matchDao = matchDao;
+    }
+
+    @Path("v1/owner/{owner}/project/{project}/stack/{stack}/z/{z}/clusteredTileBoundsForCollection/{matchCollection}")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(
+            tags = { "Section Data APIs", "Point Match APIs" },
+            value = "Get bounds for each tile with specified z sorted by connected clusters")
+    @ApiResponses(value = {
+            @ApiResponse(code = 404, message = "stack not found")
+    })
+    public List<List<TileBounds>> getClusteredTileBoundsForCollection(@PathParam("owner") final String owner,
+                                                                      @PathParam("project") final String project,
+                                                                      @PathParam("stack") final String stack,
+                                                                      @PathParam("z") final Double z,
+                                                                      @PathParam("matchCollection") final String matchCollection,
+                                                                      @QueryParam("matchOwner") final String matchOwner) {
+
+        LOG.info("getClusteredTileBoundsForCollection: entry, owner={}, project={}, stack={}, z={}, matchCollection={}",
+                 owner, project, stack, z, matchCollection);
+
+        List<List<TileBounds>> result = null;
+
+        try {
+
+            final StackId stackId = new StackId(owner, project, stack);
+            final String effectiveMatchOwner = matchOwner == null ? owner : matchOwner;
+            final MatchCollectionId matchCollectionId =
+                    MatchService.getCollectionId(effectiveMatchOwner, matchCollection);
+
+            final List<TileBounds> allTileBounds = renderDao.getTileBoundsForZ(stackId, z, null);
+
+            final Set<String> distinctSectionIds = new HashSet<>(allTileBounds.size());
+            final Map<String, TileBounds> tileIdToBoundsMap = new HashMap<>(allTileBounds.size());
+
+            allTileBounds.forEach(tb -> {
+                final String sectionId = tb.getSectionId();
+                if (sectionId != null) {
+                    distinctSectionIds.add(tb.getSectionId());
+                    tileIdToBoundsMap.put(tb.getTileId(), tb);
+                }
+            });
+
+            final Set<String> layerTileIds = tileIdToBoundsMap.keySet();
+            final TileIdsWithMatches tileIdsWithMatches = new TileIdsWithMatches();
+
+            final int numberOfSectionIds = distinctSectionIds.size();
+            if (numberOfSectionIds > 1) {
```

Comment:
Sorting works with an empty set, and one or both of the loops get skipped if `distinctSectionIds` has one or zero elements, so the `if / else` branching shouldn't be necessary here.